android/capacitor/src/main/java/com/getcapacitor/Plugin.java

```diff
@@ -489,15 +511,27 @@ public void requestPermissions(PluginCall call) {
         } else {
             // If call was made without any custom permissions, request all from plugin annotation
             if (providedPermsList == null || providedPermsList.isEmpty()) {
-                perms = new String[annotation.permissions().length];
-                for (int i = 0; i < perms.length; i++) {
-                    perms[i] = annotation.permissions()[i].permission();
+                HashSet<String> permsSet = new HashSet<>();
+                for (Permission perm : annotation.permissions()) {
+                    // If a permission is defined with no permission constants, separate it for auto-granting.
```

Comment:
I'm still confused as to why this part is necessary: `(perm.permission().length == 1 && perm.permission()[0].isEmpty())`

It is handling this case?
```
@Permission(permission = {""}, alias = ...
```

Reply:
yes or this valid definition

```
@Permission(permission = "", alias = ...
```

which on the back end translates to an array containing one empty element

Reply:
Let's switch to Kotlin.